### PR TITLE
feat: add secret management commands

### PR DESCRIPTION
**Pull request by xianml**
## What does this PR address?
add secret management commands

```
1. List
(.venv) ➜  ✗ bentoml  secret list                               
 Secret                                        Created_At                                                      Mount_As                          Keys                                                      Path                                                    
 secret-env                                    2024-05-29 09:16:12                                             env                               IS_TEST, HF_TOKEN                                         -                                                       
 secret-file7                                  2024-05-29 09:15:55                                             env                               IS_TEST                                                   -                                                       
 secret-bcvsz7                                 2024-05-29 08:42:29                                             file                              cacdac                                                    $BENTOML_HOME/aws 
2. Create 
(.venv) ➜  bentoml secret create -n secret-file8 -t file --from-file=.apikey=~/Downloads/github-recovery-codes.txt
Secret secret-file8 created successfully

3. Delete
(.venv) ➜ bentoml secret delete secret-file9                                                                     
Secret secret-file9 deleted successfully

4. Update
(.venv) ➜  bentoml secret apply s-file2 -t file --from-file key1=./auth.lua key2=xxx  -d "mount file test"
Secret s-file2 created successfully

5. Deploy with secrets

bentoml deploy -f - <<EOF
name: jtest-with-secret
bento: jtest:myuy6va6lkcbqiru
services:
  apiserver:
    scaling:
      min_replicas: 1
      max_replicas: 1
    instance_type: cpu.1
  iris_clf:
    scaling:
      min_replicas: 1
      max_replicas: 1
    instance_type: cpu.1
access_authorization: false
secrets:
  - secret-test
EOF
 
```

Fixes #(issue)

## Before submitting:

- [x] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [x] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

**Review by aarnphm — APPROVED** [review]

**Comment by frostming** [comment]
Should we merge this to 1.3 branch `dev/1.3` instead of main?

**Comment by xianml** [comment]
> Should we merge this to 1.3 branch `dev/1.3` instead of main?

Agreed.

File: src/bentoml_cli/secret.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+from bentoml.exceptions import BentoMLException
+from bentoml_cli.utils import BentoMLCommandGroup
+
+if t.TYPE_CHECKING:
+    from click import Context
+    from click import Parameter
+
+if t.TYPE_CHECKING:
+    TupleStrAny = tuple[str, ...]
+    from .utils import SharedOptions
```
Comment (frostming) [inline src/bentoml_cli/secret.py:25]: Merge these two `if t.TYPE_CHECKING`

File: src/bentoml/_internal/cloud/client.py [hunk 2]
```diff
@@ -561,6 +565,47 @@ def get_latest_model(
...
+        return schema_from_json(resp.text, SecretListSchema)
+
+    def create_secret(self, secret: CreateSecretSchema) -> SecretSchema:
+        url = urljoin(self.endpoint, "/api/v1/org_secrets")
+        resp = self.session.post(url, content=schema_to_json(secret))
+        self._check_resp(resp)
+        return schema_from_json(resp.text, SecretSchema)
+
+    def delete_secret(self, name: str):
+        url = urljoin(self.endpoint, f"/api/v1/org_secrets/name/{name}")
```
Comment (yetone) [inline src/bentoml/_internal/cloud/client.py:599]: ‌‌‌‌Expectation is a legitimate RESTful API

**Review by frostming — APPROVED** [review]

**Pull request merged**